sdk/eventgrid/azure-eventgrid/azure/eventgrid/_models.py

```diff
@@ -0,0 +1,257 @@
+        # type: (Any) -> None
+        kwargs["id"] = kwargs.get('id', uuid.uuid4())
+        kwargs["source"] = source
+        kwargs["type"] = type
+        kwargs["time"] = kwargs.get('time', dt.datetime.now(UTC()).isoformat())
+        kwargs["specversion"] = "1.0"
+
+        super(CloudEvent, self).__init__(**kwargs)
+
+
+class EventGridEvent(InternalEventGridEvent):
+    """Properties of an event published to an Event Grid topic using the EventGrid Schema.
+
+    Variables are only populated by the server, and will be ignored when sending a request.
+
+    All required parameters must be populated in order to send to Azure.
+
+    :param id: An unique identifier for the event. If not provided, one will be generated for the event.
+    :type id: str
+    :param topic: The resource path of the event source. If not provided, Event Grid will stamp onto the event.
+    :type topic: str
+    :param subject: Required. A resource path relative to the topic path.
+    :type subject: str
+    :param data: Event data specific to the event type.
+    :type data: object
+    :param event_type: Required. The type of the event that occurred.
+    :type event_type: str
+    :param event_time: The time (in UTC) of the event. If not provided, it will be the time (in UTC) the event was generated.
+    :type event_time: ~datetime.datetime
+    :ivar metadata_version: The schema version of the event metadata. If providedj, must match Event Grid Schema exactly.
+        If not provided, EventGrid will stamp onto event.
+    :vartype metadata_version: str
+    :param data_version: The schema version of the data object. If not provided, will be stamped with an empty value.
+    :type data_version: str
+    """
+
+    _validation = {
+        'id': {'required': True},
+        'subject': {'required': True},
+        'data': {'required': True},
+        'event_type': {'required': True},
+        'event_time': {'required': True},
+        'metadata_version': {'readonly': True},
+        'data_version': {'required': True},
+    }
+
+    _attribute_map = {
+        'id': {'key': 'id', 'type': 'str'},
+        'topic': {'key': 'topic', 'type': 'str'},
+        'subject': {'key': 'subject', 'type': 'str'},
+        'data': {'key': 'data', 'type': 'object'},
+        'event_type': {'key': 'eventType', 'type': 'str'},
+        'event_time': {'key': 'eventTime', 'type': 'iso-8601'},
+        'metadata_version': {'key': 'metadataVersion', 'type': 'str'},
+        'data_version': {'key': 'dataVersion', 'type': 'str'},
+    }
+
+    def __init__(self, subject, event_type, **kwargs):
+        # type: (Any) -> None
+        kwargs["id"] = kwargs.get('id', uuid.uuid4())
+        kwargs["subject"] = subject
+        kwargs["event_type"] = event_type
+        kwargs["event_time"] = kwargs.get('event_time', dt.datetime.now(UTC()).isoformat())
+        kwargs["data"] = kwargs.get('data', None)
+
+        super(EventGridEvent, self).__init__(**kwargs)
+
+class DictMixin(object):
```

Comment:
nitpicky code organization thing, I'd likely put this in a separate mixins file since it's not technically a model.